pennylane/ops/op_math/sprod.py

```diff
@@ -192,6 +192,10 @@ def eigvals(self):
     def sparse_matrix(self, wire_order=None):
         return self.scalar * self.base.sparse_matrix(wire_order=wire_order)
 
+    @property
+    def has_matrix(self):
+        return isinstance(self.base, qml.Hamiltonian) or self.base.has_matrix
+
```

Comment:
This is in `SymbolicOp`.

Reply:
In `SymbolicOp` the branch for `qml.Hamiltonian` is not included, right? So if the base is `qml.Hamiltonian`, we would report `has_matrix=False` although the `matrix` would succeed to produce a matrix.